android/test-app/src/main/java/com/arthenica/mobileffmpeg/test/SafTabFragment.java

```diff
@@ -0,0 +1,426 @@
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.text.method.ScrollingMovementMethod;
+import android.util.AndroidRuntimeException;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.MediaController;
+import android.widget.TextView;
+import android.widget.VideoView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+
+import com.arthenica.mobileffmpeg.Config;
+import com.arthenica.mobileffmpeg.ExecuteCallback;
+import com.arthenica.mobileffmpeg.FFmpeg;
+import com.arthenica.mobileffmpeg.FFprobe;
+import com.arthenica.mobileffmpeg.LogCallback;
+import com.arthenica.mobileffmpeg.LogMessage;
+import com.arthenica.mobileffmpeg.Statistics;
+import com.arthenica.mobileffmpeg.StatisticsCallback;
+import com.arthenica.mobileffmpeg.util.DialogUtil;
+import com.arthenica.mobileffmpeg.util.ResourcesUtil;
+import com.arthenica.smartexception.java.Exceptions;
+
+import java.io.File;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.concurrent.Callable;
+
+import static android.app.Activity.RESULT_OK;
+import static com.arthenica.mobileffmpeg.Config.RETURN_CODE_SUCCESS;
+import static com.arthenica.mobileffmpeg.test.MainActivity.TAG;
+import static com.arthenica.mobileffmpeg.test.MainActivity.handler;
+
+public class SafTabFragment extends Fragment {
+
+    private TextView outputText;
+    private Uri inUri;
+    private Uri outUri;
+    private static final int REQUEST_SAF_FFPROBE = 11;
+    private static final int REQUEST_SAF_FFMPEG = 12;
+
+    private boolean backFromIntent = false;
+
+    private VideoView videoView;
+    private AlertDialog progressDialog;
+    private Statistics statistics;
+
+    private Button runFFmpegButton;
+    private Button runFFprobeButton;
+
+    public SafTabFragment() {
+        super(R.layout.fragment_saf_tab);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        runFFmpegButton = view.findViewById(R.id.runFFmpegButton);
+        runFFmpegButton.setOnClickListener(new View.OnClickListener() {
+
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(Intent.ACTION_CREATE_DOCUMENT)
+                        .setType("video/*")
+                        .putExtra(Intent.EXTRA_TITLE, "video.mp4")
+                        .addCategory(Intent.CATEGORY_OPENABLE);
+                startActivityForResult(intent, REQUEST_SAF_FFMPEG);
+            }
+        });
+
+        runFFprobeButton = view.findViewById(R.id.runFFprobeButton);
+        runFFprobeButton.setOnClickListener(new View.OnClickListener() {
+
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(Intent.ACTION_OPEN_DOCUMENT)
```

Comment:
Would using `ACTION_GET_CONTENT` be a better option here? It gives us more options to choose.